content/en/platform/corda/5.0-dev-preview-2/getting-started/fast-feedback-with-the-simulator/java-flow-test-code.md

```diff
@@ -0,0 +1,93 @@
+       private MemberX500Name aliceX500 = MemberX500Name.parse("CN=Alice, OU=Test Dept, O=R3, L=London, C=GB");
+       private MemberX500Name bobX500 = MemberX500Name.parse("CN=Bob, OU=Test Dept, O=R3, L=London, C=GB");
+   ...
+   }
+   ```
+2. Declare the test method in the standard way for JUnit test, with a `@Test` annotation:
+   ```java
+   @Test
+   @SuppressWarnings("unchecked")
+   public void test_that_MyFirstFLow_returns_correct_message() {
+     ...
+   }
+   ```      
+3. Instantiate a `Simulator` class:
+   ```java
+   // Instantiate an instance of the Simulator
+   Simulator simulator = new Simulator();
+    ```
+   To find out more about this class, read the [README.md from the source code](https://github.com/corda/corda-runtime-os/blob/release/os/5.0/simulator/README.md).
+
+4. Convert the `MemberX500Name` for Alice and Bob into `HoldingIdentities` using the `HoldingIdentity` static method. The `HoldingIdentity` in Corda holds the `MemberX500Name` and the `GroupId` which is the unique identifier for the application network. However, the class used here is the simulated version provided by Simulator; that is `net.corda.simulator.HoldingIdentity` rather than `net.corda.virtualnode.HoldingEntity`.
+   ```java
+   // Create Alice's and Bob HoldingIDs
+   HoldingIdentity aliceHoldingID = HoldingIdentity.Companion.create(aliceX500);
+   HoldingIdentity bobHoldingID = HoldingIdentity.Companion.create(bobX500)
+   ```        
+5. Use Simulator to create virtual nodes for Alice and Bob from their respective holding identities. The second argument allows you to specify which flows should be loaded up to each virtual node. In this case, Alice runs the initiating flow, with Bob running the responder flow.
+   ```java
+   // Create Alice and Bob's virtual nodes, including the Class's of the flows which will be registered on each node.
+   // We don't assign Bob's virtual node to a variable because we don't need it for this particular test.
+   SimulatedVirtualNode aliceVN = simulator.createVirtualNode(aliceHoldingID, MyFirstFlow.class);
+   simulator.createVirtualNode(bobHoldingID, MyFirstFlowResponder.class);
+   ```
+   You can read more about initiating flows and responder flows in the section on [Your first flow ](../first-flow.html#initiating-and-responding-flows).
+6. Create the arguments to pass to the flow. In the flow file `MyFirstFlowStartArgs.kt`, create a class `MyFirstFlowArguments` specifically for holding the flow start arguments:
+   ```java
+   // A class to hold the arguments required to start the flow
+   public class MyFirstFlowStartArgs {
+     ...
+   }
+   ```
+   <!--no hello bob??-->
+   Use the same class to specify that the message needs to go to Bob and the message should be "Hello Bob":
+   ```java
+   // Create an instance of the MyFirstFlowStartArgs which contains the request arguments for starting the flow
+   MyFirstFlowStartArgs myFirstFlowStartArgs = new MyFirstFlowStartArgs(bobX500);
```

Comment:
@ceebee-dev can you confirm this is correct? Should the message be an argument?

Reply:
That's correct.

Reply:
I think we're saying to use the same class as created in step 6?
So:
"Use the `MyFirstFlowStartArgs` class to specify that the message should go to Bob."
?

Reply:
Yes:
"Use the MyFirstFlowStartArgs class to specify that the message should go to Bob."
or 
"Use the same class to specify that the message needs to go to Bob "
Is what we need